implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -487,4 +492,11 @@ static double[] getSloWithPositiveInf(DistributionStatisticConfig distributionSt
         return sloWithPositiveInf;
     }
 
+    static String getUserAgentHeader() {
+        if (MeterRegistry.class.getPackage().getImplementationVersion() == null) {
```

Comment:
In best practice, users should align versions (via our BOM etc.) but `MeterRegistry` is from a different module (micrometer-core) than `OtlpMeterRegistry` (micrometer-registry-otlp), so it is possible for misaligned versions to end up on the classpath. From the issue description, it sounds like it is the version of the OTLP exporter that's expected here (rather than the version of the library used for instrumentation), which I would say is the version of the micrometer-registry-otlp JAR, so probably better to get it from `OtlpMeterRegistry.class` or perhaps even better `this.getClass()` in a non-static context.

Reply:
Also, can we make this method `private`?
So together with Tommy's comment, how about this?
```java
private String getUserAgentHeader() {
    if (this.getClass().getPackage().getImplementationVersion() == null) {
        return "Micrometer-OTLP-Exporter-Java/unknown";
    }
    return "Micrometer-OTLP-Exporter-Java/" + this.getClass().getPackage().getImplementationVersion();
}
```

I'm also thinking if doing this would be better if the version number is not available:
```java
return "Micrometer-OTLP-Exporter-Java";
```